common/doom_obj_container.h

```diff
@@ -1,315 +1,199 @@
-{
-	iterator it = this->lookup_table.find(idx);
-    if (it == this->end())
-    {
-	    I_Error("Attempt to access invalid {} at idx {}", typeid(ObjType).name(), idx);
-    }
-    return it->second;
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-typename DoomObjectContainer<ObjType, IdxType, FreeFunction>::ConstObjReference DoomObjectContainer<
-    ObjType, IdxType, FreeFunction>::operator[](int idx) const
-{
-    const_iterator it = this->lookup_table.find(idx);
-    if (it == this->end())
-    {
-    	I_Error("Attempt to access invalid {} at idx {}", typeid(ObjType).name(), idx);
-    }
-    return it->second;
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-bool DoomObjectContainer<ObjType, IdxType, FreeFunction>::operator==(const ObjType* p) const
-{
-	return this->container().data() == p;
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-bool DoomObjectContainer<ObjType, IdxType, FreeFunction>::operator!=(const ObjType* p) const
-{
-	return this->container().data() != p;
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-DoomObjectContainer<ObjType, IdxType, FreeFunction>::operator const ObjType*() const
-{
-	return const_cast<ObjType>(this->container.data());
-}
-template <class ObjType, class IdxType, class FreeFunction>
-DoomObjectContainer<ObjType, IdxType, FreeFunction>::operator ObjType*()
-{
-	return this->container.data();
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-ObjType operator-(ObjType obj, DoomObjectContainer<ObjType, IdxType, FreeFunction>& container)
-{
-	return obj - container.data();
-}
-template <class ObjType, class IdxType, class FreeFunction>
-ObjType operator+(DoomObjectContainer<ObjType, IdxType, FreeFunction>& container, WORD ofs)
-{
-	return container.data() + ofs;
-}
-
-// data functions for quicker access to all objects presently stored
-
-template <class ObjType, class IdxType, class FreeFunction>
-ObjType* DoomObjectContainer<ObjType, IdxType, FreeFunction>::data()
-{
-	return this->container.data();
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-const ObjType* DoomObjectContainer<ObjType, IdxType, FreeFunction>::data() const
-{
-	return this->container.data();
-}
-
-// Capacity and Size
-
-template <class ObjType, class IdxType, class FreeFunction>
-size_t DoomObjectContainer<ObjType, IdxType, FreeFunction>::size() const
-{
-	return this->container.size();
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-size_t DoomObjectContainer<ObjType, IdxType, FreeFunction>::capacity() const
-{
-	return this->container.capacity();
-}
-
-template <class ObjType, class IdxType, class FreeFunction>
-void DoomObjectContainer<ObjType, IdxType, FreeFunction>::clear()
-{
-	for (auto& obj : this->container)
+	using LookupTable = std::unordered_map<IdxType, ObjType*>;
+	// TODO: maybe use the vector for iteration instead of the map for speed?
```

Comment:
Depends on the type of iteration. The in order table has _everything_ that has been added via initialization and dehacked. The map should always contain the working set so iterating over that is usually the most accurate of what objects are used in game.